genologics/entities.py

```diff
@@ -438,11 +439,43 @@ def create(cls, lims, container, position, **kwargs):
 
         Udfs can be sent in with the kwarg `udfs`. It should be a dictionary-like.
         """
+        instance = cls.create_in_memory_instance(lims, container, position, **kwargs)
+        data = lims.tostring(ElementTree.ElementTree(instance.root))
+        instance.root = lims.post(uri=lims.get_uri(cls._URI), data=data)
+        instance._uri = instance.root.attrib['uri']
+        return instance
+
+    @classmethod
+    def batch_create(cls, lims, instances):
+        """
+        Batch creates the samples. The list of instances is created by calls to
+        `Sample.create_request`
+        """
+        # Create a batch request from all instances:
+        batch = ElementTree.Element(nsmap(cls._PREFIX + ':details'))
+
+        for instance in instances:
+            element = copy.deepcopy(instance.root)
+            batch.append(element)
+
+        data = lims.tostring(ElementTree.ElementTree(batch))
+        response = lims.post(uri=lims.get_uri('samples/batch/create'), data=data)
+
+        ret = list()
+        for entry in response:
+            uri = entry.attrib['uri']
+            ret.append(Sample(lims=lims, uri=uri))
```

Comment:
Here, it looks like the Sample instance is just initiated with a single uri, and there is another round trip to the db when the Sample instance is to be used. Probably, there is a cache filled up in the .post() call above. I don't like this hidden-ness. I suggest not to care about this in this case :)

Reply:
Good observation. The reply from the post above actually has no info on the samples. The batch endpoint only returns the IDs of the samples. Because this call is supposed to be as fast as possible, and it's even rather unlikely that the user wants all the info about the samples created,  I think initializing it with the uri is the valid thing to do in this case, (unless we would introduce a parameter like `load_all` or similar, so the cost would be explicit to the caller).